### ISSUE TITLE
Windows binary release does not include wmlxgettext

### ISSUE THREAD

**Issue by AncientLich** [wesnoth/wesnoth#2420]
**PART I**

An user, in our italian forum, required to make translation for his own UMC, but discovered that wesnoth 1.13.x (binary release for windows) does not include the /utils folder, so wmlxgettext will be not available.
Wmlxgettext is missing there, and infact, running GUI.pyw - tab wmlxgettext - will generate this error:

`C:\Users\Arge\AppData\Local\Programs\Python\Python36-32\pythonw.exe "C:\Users\Arge\Games\Battle for Wesnoth 1.13.10\utils\wmlxgettext" --domain Wesnoth-RoH_02_The_Dwarven_Advent --directory "C:\Users\Arge\Documents\My Games\Wesnoth1.13\data\add-ons\RoH_02_The_Dwarven_Advent" --recursive -o "C:\Users\Arge\Documents\My Games\Wesnoth1.13\data\add-ons\skele" --no-text-colors
C:\Users\Arge\AppData\Local\Programs\Python\Python36-32\pythonw.exe: can't open file 'C:\Users\Arge\Games\Battle for Wesnoth 1.13.10\utils\wmlxgettext': [Errno 2] No such file or directory`

So the binary version for windows (and, probably, for Mac Os) should include wmlxgettext somehow (moving it to /data/tools, other solutions?)

-------
**PART II**

wmlxgettext, can now be runned in windows. But windows, as you know, has a limited command length. So this is why the '--recursive' option is likely "mandatory" on windows to allow a brief command line, like this one
`python3 wmlxgettext -o outfile.po --directory=UMC_DIRECTORY --domain=YOUR_DOMAIN --recursive`
so wmlxgettext will scan all the UMC directory, without having to explicit file list (with a lot of space saved from command line length).
But how to prevent (or to warn), on windows, the user about this problem?
1) Perhaps a warn or an hint showed in the GUI if runned on windows? (so the hint could be seen by the end-user as soon as the wmlxgettext tag was selected, without having to discovering it later)
2) should I "force" windows users to use --recursive (generating an error if, under windows, an user try to list the files to scan)?
3) should I add a simple warn that is visible under windows (but in this case, if I have a truncated command line, wmlxgettext could be not able to show the warning... becouse if the truncated line creates a bad command line, python argparse will manage it as a wrong command line: the help menu about how to use command line will be showed)

-------
I'd like to hear your opinions about all those two "parts" of this report ;)

**Comment by loonycyborg** [comment]
I think we should make another dir for umc dev tools. There are mostly packaging and server maintenance scripts currently in utils/. For now users can check out that dir from git if binary release they're using isn't including tool they need.

**Comment by AncientLich** [comment]
A good point. However wmlxgettext is both packaging script (for po-update target build, extracting all .po for core translations) and also an umc tool. The dedicated directory, if well managed, could be a solution if also build system (scons, cmake) will be update to find the new location of wmlxgettext. However, if we plan to move wmlxgettext location, perhaps the /data/tools could be the best location (becouse GUI.pyw, wmllint and other utilities that GUI.pyw uses are there). However I will agree with any solution you develop you will think it is best.
Any hint also about "part II" ?

**Comment by CelticMinstrel** [comment]
Speaking of GUI.pyw, does that gracefully handle the case where wmlxgettext is not present?

**Comment by loonycyborg** [comment]
wrt 2. I don't think the tool should have different behavior on different oses. People are unlikely to blow command line limit if using it from command line and running it only on one particular file might still be needed.

**Comment by AncientLich** [comment]
Yeah... I agree that the tool should work the same way regardless of OS.
For this reason I would personally suggest to add a warning (a pop-up message, for example) in GUI.pyw (and not in wmlxgettext) wich will warn the user that, for translating an entire UMC under windows, it is suggested to use 'recursive' option to avoid possible problems related to limited command length in windows (or something similar).
So the user will be able to make its decision. Using --recursive is an alternate syntax (thinked to make wmlxgettext easier to use) but I would dislike too to transform it in a mandatory syntax under windows (it would be "break" pot-update target on windows, since --recursive cannot be used together with explicit filelist) and, right point, that solution would limit the tool usage, for example forbidding to extract .po from a single WML/lua file.

**Comment by GregoryLundberg** [comment]
Sorry, I saw the add to a past milestone and didn't see the close.
